### ISSUE TITLE
Prefferential Displacement in tp.locate

### ISSUE THREAD

**Issue by praptibajaj28** [soft-matter/trackpy#562]
Dear Trackpy Community,

I am beginning to use trackpy for tracking micrometer-szied particles in plasma (complex plasmas). The simulated image is given here: 
![oi_20190617](https://user-images.githubusercontent.com/54531209/63680211-6c54fc00-c7f3-11e9-878c-9cb7901bf93c.png)
 . Using this image with 787 simulated particles, trackpy could find 786 of these particles, as shown here: 
![Figure_sim_trackpy](https://user-images.githubusercontent.com/54531209/63680235-7d057200-c7f3-11e9-89e6-378cfce75bc3.png)
. However, when I compare the positions that have been simulated v/s positions that are found (i.e. nearest neighbour using simulated position minus found position), I find a prefferential displacement in the found positions. Most of the particles are found on the left side of the exact position as simulated. 

I belive this could be due to the funtion "where_close" but am not entirely sure of why this is happening and how to solve this issue. I have tried increasing and decreasing "max_iterations" while calling "tp.batch". I have also tried usng different engine options in this function. However, nothing seems to change this statistics: 
1. number of particles found with positive x  displacement compared to simulated particle position:  248
2. number of particles found in negative x :  538
3. number of particles found in positive y :  410
4. number of particles found in negative y :  376
 Here is the plot to show this statistic : 
![image](https://user-images.githubusercontent.com/54531209/63680067-12543680-c7f3-11e9-9ddf-0997311548e4.png)

I will be highly obliged for any help in this direction. Thank you very much.

Best regards
Prapti

**Comment by caspervdw** [comment]
Hi,
Thanks for the report! I wonder how you generate your artificial images? You have to put them at random sub pixel locations. And what is the average bias? 

A source of bias might be the boxcar filter in the preprocessing. 

I wrote an article about sources of bias and imprecision in feature finding: if you’re interested, please see https://arxiv.org/pdf/1607.08819

**Comment by praptibajaj28** [comment]
Dear all, 

I could finally find out the problem. The simulation software I was using (called staub, designed for complex plasmas only) was not distributing intensities spherically as my particles are very small. This is why I observed this prefferential displacement. This displacement effect disappeared when I increased the size of my particles and saw uniform intensity distribution in each particle (in x-axis, this disappeared but there were still particles displaced in +ve y-axis which might be due to wrong coordinates given by the simulation software?). Is there any other software that someone could suggest to get some simulated image with a particular particle size (around 1 sigma), 1000 particles  in 640x237 pixels image size and a user-given image background? It would be very interesting for my project to create artificial experimental images and then compare the efficiency of particle detection and tracking of different toolkits like trackpy, staub and others available for these purposes. 

Thank you very much.
